frontend/components/MainPage/header/AgentButton/AgentNotRunningButton.tsx

```diff
@@ -32,10 +33,57 @@ import { AgentConfig } from '@/types/Agent';
 import { delayInSeconds } from '@/utils/delay';
 import { updateServiceIfNeeded } from '@/utils/service';
 
+/**
+ * Health check hook
+ */
+const useHealthCheck = () => {
+  const { selectedAgentType } = useServices();
+  const { goto: gotoPage } = usePageState();
+
+  return useCallback(
+    async (serviceId?: string) => {
+      if (!serviceId) return true; // TODO: check
+
+      try {
+        const response = await ServicesService.getDeployment(serviceId);
+        const healthcheck = response.healthcheck;
+        if (isEmpty(healthcheck) || !healthcheck.env_var_status?.needs_update) {
+          return true;
+        }
+
+        if (selectedAgentType !== AgentType.Memeooorr) {
+          return true;
+        }
+
+        const reason =
+          healthcheck.env_var_status?.env_vars?.['TWIKIT_COOKIES']?.reason;
```

Comment:
@dvilelaf correct me if I'm wrong, but it's "TWIKIT_USERNAME" we should check instead? based on the agent's code
<img width="747" alt="image" src="https://github.com/user-attachments/assets/4fd29f71-014f-47ea-abeb-01269db7ef5c" />